src/test/java/spoon/MavenLauncherTest.java

```diff
@@ -208,12 +208,11 @@ public void mavenLauncherTestMultiModulesAndVariables() {
 	}
 
 	@Test
-	public void testGuessMavenHome() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
-		Method method = SpoonPom.class.getDeclaredMethod("guessMavenHome", new Class[]{});
-		method.setAccessible(true);
-		String mvnHome = (String) method.invoke(null, new Object[]{});
-		File mvnDir = new File(mvnHome);
-		assertTrue(mvnDir.exists());
-		assertTrue(mvnDir.isDirectory());
+	public void testGuessMavenHome() {
+		// contract: it should correctly fetch path to maven home
+		String pathToMavenHome = SpoonPom.guessMavenHome();
+		File mavenHome = new File(pathToMavenHome);
+		assertTrue(mavenHome.exists());
```

Comment:
Can you add here a String explaining what failed? The error message of `assertTrue` is kinda bad.

Reply:
I throw a `SpoonException` in `guessMavenHome` (message: Couldn't find path to maven home.). We will never reach this line if the test fails.

Is this the correct to way to write this test because it doesn't make sense to `assertDoesNotThrow`?

Reply:
Agree, assertTrue with message is a best practice, but in this case we can proceed.

Keeping trace of this in good first issue #4307

Reply:
I've been thinking to introduce the full range of Hamcrest matchers into Spoon, they truly bring some wonderful expressiveness to tests. The perfect matcher here would be `aFileNamed()`.

Reply:
Completely agree. Moreover, it would also help refactor our tests. We currently use Junit4, Junit5, and hamcrest. It's best to have just one library handling it all.

Reply:
Hamcrest is just the matchers, it's not a testing framework. So, it would be Junit5 + Hamcrest matchers.